lms/djangoapps/completion/api/v1/tasks.py

```diff
@@ -0,0 +1,29 @@
+import logging
+from celery import current_task, task  # pylint: disable=no-name-in-module
+from celery.exceptions import RetryTaskError  # pylint: disable=no-name-in-module, import-error
+from celery.states import FAILURE, RETRY, SUCCESS  # pylint: disable=no-name-in-module, import-error
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from student.models import CourseEnrollmentManager
+from lms.djangoapps.completion.utils import GenerateCompletionReport
+
+log = logging.getLogger('celery.worker.consumer')
+
+
+@task(default_retry_delay=5, max_retries=5)
+def generate_report(params):
+
+    course_id = params[0]
```

Comment:
why don't you call the params explicitly? "def generate_report(course_id):"

Reply:
according to this: http://docs.celeryproject.org/en/3.1/userguide/calling.html#example, you should be able to call the task with args and kwargs